src/mpris_event_handler.rs

```diff
@@ -1151,10 +1151,16 @@ impl MprisTask {
             },
             PlayerEvent::Unavailable { .. } => { },
             PlayerEvent::VolumeChanged {
-                // volume
+                volume,
                 ..
             } => {
-                // TODO: Handle volume
+                let iface_ref = self.mpris_player_iface().await;
+                let mut iface = iface_ref.get_mut().await;
+                let mut swapped_volume = volume;
+                std::mem::swap(&mut iface.volume, &mut swapped_volume);
+                if iface.volume != swapped_volume {
+                    iface.volume_changed(iface_ref.signal_context()).await?;
+                }
```

Comment:
Wouldn't this be simpler as

```suggestion
            PlayerEvent::VolumeChanged {
                new_volume,
                ..
            } => {
                let iface_ref = self.mpris_player_iface().await;
                let mut iface = iface_ref.get_mut().await;
                if iface.volume != new_volume {
                    iface.volume = new_volume;
                    iface.volume_changed(iface_ref.signal_context()).await?;
                }
```

Reply:
Obviously! Don't know what I had in mind while doing this. Thanks.